### DISCUSSION TITLE
Validation .ForceValidate(); from ViewModel

### DISCUSSION THREAD

**Discussion by Eden-Mor** [xamarin/XamarinCommunityToolkit#1640] [Q&A] [answered]
Hello, 
Me and a couple of other people from the DotNetEvolution discord are trying to achieve validation re-check from the ViewModel and were unable to figure out how.

I've included a sample (im unable to upload .zip files to discussion) 
In the first page of the sample (AboutPage)
Theres an Entry with the x:Name="NameEntryValidator"
We want to NameEntryValidator.ForceValidate(); from the view model.
This solution isnt very MVVM friendly but its just an example.
The sample includes an example (do not click on the entry field)
The entry has a required 3 minimum length field for example, and the field was already supplied with 2 characters.
It has an IsValid binding, the binding should Fail, but it does not.

The next page leading to it (from Test Button) has a neat solution
Validation can be bound into groups
And group bindings can invoke the ForceValidate(); method.

Thank you!

https://www.dropbox.com/s/d6uohi9zihoru74/ValidationForceCheck.zip?dl=0

**Comment by AndreiMisiukevich** [comment, marked as answer]
>> im unable to upload .zip files to discussion
Can you please create a git repo then? (I don't have a dropbox account and cannot download it, sorry)

>> We want to NameEntryValidator.ForceValidate(); from the view model.

View
```xaml
<TextValidationBehavior ForceValidateCommand={Binding ForceValidateCommand}>
...
```

ViewModel
```csharp
public ICommand ForceValidateCommand { get; set; }
```

Usage
```csharp
ForceValidateCommand?.Execute(null);
```

**Reply by vhugogarcia** [reply]
> > > im unable to upload .zip files to discussion
> > > Can you please create a git repo then? (I don't have a dropbox account and cannot download it, sorry)
> 
> > > We want to NameEntryValidator.ForceValidate(); from the view model.
> 
> View
> 
> ```
> <TextValidationBehavior ForceValidateCommand={Binding ForceValidateCommand}>
> ...
> ```
> 
> ViewModel
> 
> ```cs
> public ICommand ForceValidateCommand { get; set; }
> ```
> 
> Usage
> 
> ```cs
> ForceValidateCommand?.Execute(null);
> ```

I just tried your solution, however, the ```ForceValidateCommand?.Execute(null);``` does not return true or false to idenfity if the validation result was OK or no.

For example if you have an entry like this:

```
<controls:StandardEntry
                        x:Name="Email"
                        Margin="0,0,0,0"
                        Padding="30,10,30,10"
                        BackgroundColor="#E7E7E7"
                        BorderColor="#E7E7E7"
                        BorderThickness="1"
                        CornerRadius="30"
                        FontFamily="NunitoRegular"
                        FontSize="Medium"
                        HeightRequest="60"
                        Keyboard="Email"
                        MaxLength="120"
                        Placeholder="{xct:Translate GeneralInputPlaceholderEmail}"
                        PlaceholderColor="#888888"
                        Text="{Binding Email, Mode=TwoWay}"
                        TextColor="#000000">
                        <Entry.Behaviors>
                            <xct:EmailValidationBehavior
                                x:Name="EmailValidator"
                                DecorationFlags="Trim"
                                Flags="ValidateOnUnfocusing"
                                ForceValidateCommand="{Binding ForceValidateCommand}"
                                IsValid="{Binding EmailValid}" />
                        </Entry.Behaviors>
                    </controls:StandardEntry>
```

On the Button command even if you force the validation on the MVVM it will continue the execution and there is no way to stop the execution.

**Reply by Eden-Mor** [reply]
About not being able to download the project in dropbox,
You just have to hit the x button where it asks you to log in and download from the top right corner.

>I just tried your solution, however, the ForceValidateCommand?.Execute(null); does not return true or false to idenfity if the validation result was OK or no.

ForceValidate is suppose to update the IsValid field, not return a bool of isvalid
From what I understand you need to ForceValidate and then check your IsValid binding.
I have yet to try but I will go check soon.



Edit: this fixes my issue! thanks @AndreiMisiukevich.
Im going to keep this open for @vhugogarcia.

**Reply by AndreiMisiukevich** [reply]
Yes, the command doesn't return result. you have to bind `IsValid` as well

**Reply by vhugogarcia** [reply]
> About not being able to download the project in dropbox,
> 
> You just have to hit the x button where it asks you to log in and download from the top right corner.
> 
> 
> 
> >I just tried your solution, however, the ForceValidateCommand?.Execute(null); does not return true or false to idenfity if the validation result was OK or no.
> 
> 
> 
> ForceValidate is suppose to update the IsValid field, not return a bool of isvalid
> 
> From what I understand you need to ForceValidate and then check your IsValid binding.
> 
> I have yet to try but I will go check soon.
> 
> 
> 
> 
> 
> 
> 
> Edit: this fixes my issue! thanks @AndreiMisiukevich.
> 
> Im going to keep this open for @vhugogarcia.

oh! That makes sense. Let me give it a try. 

Thank you all for your feedback

**Reply by vhugogarcia** [reply]
I can confirm it worked as expected yeah! 

thanks @Xenooooon for posting the issue and confirming it. Also thanks @AndreiMisiukevich for sharing how to implement it.

**Comment by vhugogarcia** [comment]
I can confirm it worked as expected yeah!

thanks @Xenooooon for posting the issue and confirming it. Also thanks @AndreiMisiukevich for sharing how to implement it.
